containerwebapp/entrypoint.sh

```diff
@@ -43,28 +43,20 @@ fi
 
 echo "Resource Group Name: ${RESOURCE_GROUP_NAME}"
 
-AZCLI_ARGUMENT=" --docker-custom-image-name ${CONTAINER_IMAGE_NAME}"
-
-if [[ ! -z $CONTAINER_IMAGE_TAG ]];
-then
-    AZCLI_ARGUMENT="${AZCLI_ARGUMENT}:${CONTAINER_IMAGE_TAG}"
-fi
-
-if [[ ! -z $DOCKER_USERNAME && ! -z $DOCKER_PASSWORD ]];
-then
-    AZCLI_ARGUMENT="${AZCLI_ARGUMENT} --docker-registry-server-user ${DOCKER_USERNAME} --docker-registry-server-password ${DOCKER_PASSWORD}"
-fi
-
-if [[ ! -z $DOCKER_REGISTRY_URL ]]
+if [[ -n $CONTAINER_IMAGE_TAG ]];
 then
-    AZCLI_ARGUMENT="${AZCLI_ARGUMENT} --docker-registry-server-url ${DOCKER_REGISTRY_URL}"
+    CONTAINER_IMAGE_NAME="${CONTAINER_IMAGE_NAME}:${CONTAINER_IMAGE_TAG}"
 fi
 
 echo "Initiating Container deployment..."
 
-az webapp config container set -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" $AZCLI_ARGUMENT
+az webapp config container set -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" \
+    --docker-custom-image-name "${CONTAINER_IMAGE_NAME}" \
+    --docker-registry-server-user "${DOCKER_USERNAME}" \
+    --docker-registry-server-password "${DOCKER_PASSWORD}" \
```

Comment:
handle empty values explicitly and dont rely on az handling as we dont control it.

Reply:
done